dspace-server-webapp/src/test/java/org/dspace/app/rest/DiscoveryRestControllerIT.java

```diff
@@ -3717,4 +3719,195 @@ public void discoverSearchObjectsTestForWithdrawnOrPrivateItemsByAdminUser() thr
+        // With a query stating 'Test'
+
+        String adminToken = getAuthToken(admin.getEmail(), password);
+
+        getClient(adminToken).perform(get("/api/discover/search/objects")
+                .param("configuration", "discoverableAndUndiscoverableItems")
+                .param("query", "Test"))
+
+                //** THEN **
+
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.type", is("discover")))
+                .andExpect(jsonPath("$._embedded.searchResult.page", is(
+                        PageMatcher.pageEntryWithTotalPagesAndElements(0, 20, 1, 3)
+                )))
+                .andExpect(jsonPath("$._embedded.searchResult._embedded.objects",
+                        Matchers.containsInAnyOrder(
+                                SearchResultMatcher.matchOnItemName("item", "items", "Public Test Item"),
+                                SearchResultMatcher.matchOnItemName("item", "items", "Withdrawn Test Item"),
+                                SearchResultMatcher.matchOnItemName("item", "items", "Private Test Item")
+                        )
+                ))
+                .andExpect(jsonPath("$._embedded.facets", Matchers.hasItems(
+                        allOf(
+                                hasJsonPath("$.name", is("discoverable")),
+                                hasJsonPath("$._embedded.values", Matchers.hasItems(
+                                        allOf(
+                                                hasJsonPath("$.label", is("true")),
+                                                hasJsonPath("$.count", is(2))
+                                        ),
+                                        allOf(
+                                                hasJsonPath("$.label", is("false")),
+                                                hasJsonPath("$.count", is(1))
+                                        )
+                                ))
+                        ),
+                        allOf(
+                                hasJsonPath("$.name", is("withdrawn")),
+                                hasJsonPath("$._embedded.values", Matchers.hasItems(
+                                        allOf(
+                                                hasJsonPath("$.label", is("true")),
+                                                hasJsonPath("$.count", is(1))
+                                        ),
+                                        allOf(
+                                                hasJsonPath("$.label", is("false")),
+                                                hasJsonPath("$.count", is(2))
+                                        )
+                                ))
+                        )
+                )))
+                .andExpect(jsonPath("$._links.self.href", containsString("/api/discover/search/objects")));
+    }
```

Comment:
can we have a test for the withdrawn or private facets as they are quite special? or are such test already in place?

Reply:
The facet results are tested, performing them as filters is not tested. Since they are not metadata facets, we can indeed add them as well